src/main/java/net/greatstart/controllers/PasswordResetController.java

```diff
@@ -1,117 +1,79 @@
 import org.springframework.beans.factory.annotation.Qualifier;
 import org.springframework.context.MessageSource;
-import org.springframework.security.access.prepost.PreAuthorize;
-import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
 import org.springframework.security.crypto.password.PasswordEncoder;
-import org.springframework.stereotype.Controller;
-import org.springframework.validation.Errors;
+import org.springframework.web.bind.annotation.ExceptionHandler;
 import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.ModelAttribute;
 import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestParam;
-import org.springframework.web.servlet.ModelAndView;
+import org.springframework.web.bind.annotation.RestController;
 
 import javax.servlet.http.HttpServletRequest;
-import javax.validation.Valid;
 import java.util.Locale;
 
-@Controller
+
+@RestController
 public class PasswordResetController {
 
-    public static final String MESSAGE = "message";
     private PasswordEncoder passwordEncoder;
     private MessageSource messages;
     private UserService userService;
     private SecurityService securityService;
-    private PasswordValidationService validationService;
     private MailService mailService;
 
     @Autowired
     public PasswordResetController(@Qualifier("messageSource") MessageSource messages, UserService userService,
                                    SecurityService securityService, PasswordEncoder passwordEncoder,
-                                   PasswordValidationService validationService, MailService mailService) {
+                                   MailService mailService) {
         this.messages = messages;
         this.userService = userService;
         this.securityService = securityService;
         this.passwordEncoder = passwordEncoder;
-        this.validationService = validationService;
         this.mailService = mailService;
     }
 
     @GetMapping(value = "/user/resetPassword")
-    public ModelAndView showForgotPassForm() {
-        ModelAndView model = new ModelAndView("user/forgotPassword");
-        model.addObject("user", new DtoUser());
-        return model;
-    }
-
-    @PostMapping(value = "/user/resetPassword")
-    public ModelAndView processEmail(HttpServletRequest request, @Valid @ModelAttribute("user") DtoUser dtoUser, Errors
-        errors) {
-        ModelAndView model = new ModelAndView("user/forgotPassword");
-        if (errors.hasErrors()) {
-            return model;
-        }
-        User user = userService.getUserByEmail(dtoUser.getEmail());
+    public ResponseEntity<String[]> processEmail(HttpServletRequest request, @RequestParam("email") String email) {
```

Comment:
1. No need to indicate param name if it has the same name as imported variable
2. Why do you return String[]?